### PR TITLE
Implemented configurable search delay

### PR DESCRIPTION
**Pull request by martikyan**
Implemented configurable search delay

#599

### CODE REVIEW

File: Maccy/MenuHeader/MenuHeaderView.swift [hunk 1]
```diff
@@ -118,10 +118,15 @@ class MenuHeaderView: NSView, NSSearchFieldDelegate {
   }
 
   private func fireNotification() {
+    searchThrottler.minimumDelay = TimeInterval(Double(UserDefaults.standard.searchDelay) / 1000)
```
Comment (p0deje) [inline Maccy/MenuHeader/MenuHeaderView.swift:121]: Are you doing this to ensure new values are picked up when the user changes delay? If yes, I'd rather have an observer updating delay whenever the UserDefaults value changes. There are examples of this in `Maccy.initializeObservers()`.

**Review by p0deje — COMMENTED** [review]
Thank you for the pull request. I haven't pulled this to test yet, but I can already see that localizations need to be added before this can be released. If you would like to add those, you will need to:

1. Install bartycrouch from Homebrew.
2. Update localization files by running 
```
bartycrouch update
```
3. It would generate changes in other files as well - please revert those.
4. Go though all new strings in Appearance preferences localization files and update them manually (I use Google Translate for that).

**Comment by martikyan** [comment]
@p0deje thanks for the quick response
I have made the changes you requested. Translated strings using ChatGPT, hope this will work as expected.

That was a huge manual work

**Comment by p0deje** [comment]
@martikyan I am overwhelmed currently at work, but I will definitely get to review/merge your PR, sorry for the delay.

**Comment by p0deje** [comment]
Sorry for taking so long to review, I'm currently trying to test this, but I keep getting weird results when used with large values (e.g. 5000 ms). I type "hello" and some of the keys are just dropped so the search would contain just "el". Do you experience the same problem?

**Comment by martikyan** [comment]
> I type "hello" and some of the keys are just dropped so the search would contain just "el". Do you experience the same problem?

Hmm, I couldn't reproduce this. What's your clipboard size? Do you have specific steps to be able to make it happen?

**Comment by p0deje** [comment]
@martikyan It might be somehow related to Sonoma - can you try on 0.27.0?

**Comment by martikyan** [comment]
> can you try on 0.27.0?

Sorry, I use my Mac mainly for non-Mac development so I don't install preview software :(

**Comment by p0deje** [comment]
> Sorry, I use my Mac mainly for non-Mac development so I don't install preview software :(
I meant Maccy 0.27.0 - just released it.

**Comment by martikyan** [comment]
Tried on 0.27.0, cannot reproduce.
Have you noticed that sometimes when you open Maccy, click on the search bar before typing, everything you type is selected. Is it the expected behavior? Can that be somehow related to this issue?

**Comment by martikyan** [comment]
(Unrelated to the issue) @p0deje Before merging this, let's consider having a button cell to enable/disable instant search instead of text input and stepper. What do you think about that?

**Comment by p0deje** [comment]
> Tried on 0.27.0, cannot reproduce.
> Have you noticed that sometimes when you open Maccy, click on the search bar before typing, everything you type is selected. Is it the expected behavior? Can that be somehow related to this issue?

Just noticed this - I think it's a bug but I don't think it's related.

> Sorry, I use my Mac mainly for non-Mac development so I don't install preview software :(

Sonoma should be out tomorrow - are you planning to upgrade?

**Comment by martikyan** [comment]
> are you planning to upgrade

Sure will do

**Comment by martikyan** [comment]
@p0deje I was able to reproduce that after upgrading to Sonoma and it got fixed as soon as I back-merged from master.

But now I experience other kind of issue, when setting the update interval to 0 and typing fast opening closing Maccy randomly, the results will disappear at some moment with no header

image

After this, keyboard shortcut to open Maccy's popup won't work anymore. I don't know what broke that. Any ideas what can cause that?

**Comment by martikyan** [comment]
> ... typing fast opening closing Maccy randomly, the results will disappear at some moment with no header

@p0deje Reproduced this on 0.27.0 version as well. Will open a separate issue when there'll be more info

**Comment by p0deje** [comment]
> But now I experience other kind of issue, when setting the update interval to 0 and typing fast opening closing Maccy randomly, the results will disappear at some moment with no header

Yeah, that's me trying to workaround Sonoma bugs https://github.com/p0deje/Maccy/commit/e71736adc911e7cd88ce287ee10544270bfa4388. I hope I'll find a better solution that prevents header from disappearing.

**Comment by p0deje** [comment]
@martikyan Would you mind rebasing the PR onto the latest master? There are quite a lot of conflicts due to Preferences -> Settings package move.

**Comment by martikyan** [comment]
> @martikyan Would you mind rebasing the PR onto the latest master? There are quite a lot of conflicts due to Preferences -> Settings package move.

@p0deje Let's have this PR abandoned, and just decrease the refresh time from 400ms to something like 100ms. I don't think configuring the refresh time will make this tool more handy rather than more complex. If set to <100ms, when typing in the search bar, the previous search requests won't be cancelled and the app will start lagging. 400ms is too slow for the updates.
What do you think?

**Comment by p0deje** [comment]
> @p0deje Let's have this PR abandoned, and just decrease the refresh time from 400ms to something like 100ms. I don't think configuring the refresh time will make this tool more handy rather than more complex. If set to <100ms, when typing in the search bar, the previous search requests won't be cancelled and the app will start lagging. 400ms is too slow for the updates.
What do you think?

@martikyan With the recent update to Sonoma the search got noticeably slower and I am not sure 100ms would be a good choice. Can you try playing with the delay in Sonoma with 999 history items and see what is the smallest delay that doesn't create a noticeable log?

**Pull request closed without merging**